scripts/A16.R

```diff
@@ -0,0 +1,76 @@
+# create ds object --------------------------------------------------------
+
+ds <- create_dataset(id = "A16")
+
+
+# download the data -------------------------------------------------------
+
+ds <- download_data(ds)
+
+# data cleaning -----------------------------------------------------------
+
+### Filter specific dimensions
+
+# To reduce the dataset size, we will not include sex and citizenship category
+ds$data %>%
+  janitor::clean_names() %>%
+  dplyr::filter(
+    sex == "Sex - total" &
+      citizenship_category == "Citizenship (category) - total"
+  ) %>%
+  dplyr::select(-sex, -citizenship_category) -> ds$data
+
+# Pivot indicators into 1 indicator per column and cleanup names
+ds$data %>%
+  tidyr::pivot_wider(
+    names_from = demographic_component,
+    values_from = demographic_balance_by_canton
+  ) %>%
+  janitor::clean_names() %>%
+  dplyr::rename(
+    "total_population" = population_on_1_january,
+    "births" = live_birth,
+    "deaths" = death,
+    "net_migration" = net_migration_incl_change_of_population_type,
+    "immigration" = immigration_incl_change_of_population_type
+  ) -> ds$data
+
+# Remove redundant or constant columns
+# + acquisition of swiss citizenship is always 0
+# + The 'change of population type' component is
+#   always included in the demographic components of
+#   'immigration' and 'net migration'.
+ds$data %>%
+  dplyr::select(
+    -change_of_population_type,
+    -population_on_31_december,
+    -natural_change,
+    -acquisition_of_swiss_citizenship
```

Comment:
I would leave in `acquisition_of_swiss_citizenship` since this is interesting. Redundance is not an issue here, where you want to translate natural language question into sql.  What ever term might resemble a natural language question should stay. Also you need to consider that the data in these tables is never complete.

Reply:
This is interesting, but the way this is coded in a way that's unintuitive. I'll make sure to write queries that explicitely use it.

```
   year  canton                 acquisition_of_swiss_citi…¹ citizenship_category
   <chr> <chr>                                        <dbl> <chr>               
 1 1971  Aargau                                           0 Citizenship (catego…
 2 1971  Aargau                                         746 Switzerland         
 3 1971  Aargau                                        -746 Foreign country     
 4 1971  Appenzell Ausserrhoden                           0 Citizenship (catego…
 5 1971  Appenzell Ausserrhoden                          73 Switzerland         
 6 1971  Appenzell Ausserrhoden                         -73 Foreign country     
 7 1971  Appenzell Innerrhoden                            0 Citizenship (catego…
 8 1971  Appenzell Innerrhoden                           24 Switzerland         
 9 1971  Appenzell Innerrhoden                          -24 Foreign country     

```